### DISCUSSION TITLE
Connect to server without query parameters

### DISCUSSION THREAD

**Discussion by vbaptistella** [socketio/socket.io#4021] [Q&A]
I am currently working on a project that is already in production using socket.io, and it needs to migrate to a gateway that blocks any type of query string and rejects the connection.

I could only find information about adding more parameters on the docs, but is there a way to make socket.io work without (or remove) the default query parameters (EIO, transport and t)?

**Comment by darrachequesne** [comment]
No, there is currently no way to do this.

> it needs to migrate to a gateway that blocks any type of query string

That's... interesting. Why is that?

**Reply by vbaptistella** [reply]
It's a corporate gateway that the company I work at decided to use, and they won't allow query strings due to security concerns (according to them, allowing query strings make the server susceptible to DDoS)

**Comment by jerryOluranti** [comment]
I also have the same issue here
@darrachequesne can you please tell the reason why this isn't possible?

> I am currently working on a project that is already in production using socket.io, and it needs to migrate to a gateway that blocks any type of query string and rejects the connection.

**Comment by Ark-kun** [comment]
@darrachequesne 
We have same issue. The Kubernetes API Proxy strips all query parameters from WebSockets requests: https://github.com/kubernetes/kubernetes/issues/77142
